packages/tools/pluggable-widgets-tools/scripts/e2e.js

```diff
@@ -49,51 +60,117 @@ async function main() {
+                `-u root -e MENDIX_VERSION=${latestMendixVersion} --entrypoint /bin/bash ` +
+                `mendix/runtime-base:${latestMendixVersion}-rhel /shared/runtime.sh`
+        ).toString();
+        await waitForAvailability(`http://localhost:${runtimePort}`, "runtime");
 
-    let attempts = 60;
-    for (; attempts > 0; --attempts) {
-        try {
-            const response = await fetch(`http://localhost:${freePort}`);
-            if (response.ok) {
-                break;
-            }
-        } catch (e) {
-            console.log(`Could not reach http://localhost:${freePort}, trying again...`);
-        }
-        await new Promise(resolve => setTimeout(resolve, 3000));
-    }
+        if (isNativeEnabled) {
+            const nativeAppPaths = await downloadNativeApps();
 
-    try {
-        if (attempts === 0) {
-            throw new Error("Runtime didn't start in time, existing now...");
+            // Android starts
+            mkdir("-p", "res/raw");
+            touch("res/raw/runtime_url");
+            execSync(`echo http://10.0.2.2:${runtimePort} > res/raw/runtime_url`);
+            execSync(`zip -u ${nativeAppPaths.androidPath} res/raw/runtime_url`);
+
+            mkdir("-p", "assets");
+            cp(
+                join(
+                    process.cwd(),
+                    "tests/testProject",
+                    "deployment/native/bundle/android/assets/index.android.bundle"
+                ),
+                "assets"
+            );
+
+            execSync(`zip -ur ${nativeAppPaths.androidPath} assets`);
+            execSync(`zip -ur ${nativeAppPaths.androidPath} res`);
+
+            execSync(
+                `echo android | $ANDROID_HOME/build-tools/$ANDROID_BUILD_TOOLS/apksigner sign --ks ../../tools/pluggable-widgets-tools/src/native/debug.keystore ${nativeAppPaths.androidPath}`
+            );
+
+            // Resign the binary because it needs to be signed with the same signer as real apk
+            execSync(
+                `echo android | $ANDROID_HOME/build-tools/$ANDROID_BUILD_TOOLS/apksigner sign --ks ../../tools/pluggable-widgets-tools/src/native/debug.keystore ${nativeAppPaths.androidTestBinaryPath}`
+            );
+
+            console.warn(nativeAppPaths.androidPath);
+            execSync(`export TEST_NATIVE_APP_ANDROID=${nativeAppPaths.androidPath}`);
+            execSync(`export TEST_NATIVE_APP_ANDROID_TEST_BINARY=${nativeAppPaths.androidTestBinaryPath}`);
+
+            // Android ends
+
+            execSync(`detox test --configuration android`, {
+                stdio: "inherit",
+                env: {
+                    ...process.env,
+                    TEST_NATIVE_APP_ANDROID: nativeAppPaths.androidPath,
+                    TEST_NATIVE_APP_ANDROID_TEST_BINARY: nativeAppPaths.androidTestBinaryPath
+                }
+            });
+
+            // Execute ios detox
+            if (!process.env.TRAVIS) {
```

Comment:
i feel that we should do checks for ios/android based on something more reliable than TRAVIS